From 696151ca382c7b166d349f704812dd73bc2fa7a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hannes=20M=C3=BCller?= <>
Date: Sat, 21 Dec 2024 19:20:17 +0100
Subject: [PATCH] libguile/scm.h: fix BUILDING_LIBGUILE for -Werror=undef

BUILDING_LIBGUILE is not always defined. This is
signaled using -Werror=undef in code using libguile.

This fixes commit dc3a3a84f908f4a16e95a2c3bb412861521960dc
---
 libguile/scm.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libguile/scm.h b/libguile/scm.h
index 4d079b1a8..180b40159 100644
--- a/libguile/scm.h
+++ b/libguile/scm.h
@@ -718,9 +718,9 @@ enum scm_tc8_tags
 
 /* SCM_API is a macro prepended to all function and data definitions
    which should be exported from libguile. */
-#if BUILDING_LIBGUILE && HAVE_VISIBILITY
+#if defined BUILDING_LIBGUILE && HAVE_VISIBILITY
 # define SCM_API extern __attribute__((__visibility__("default")))
-#elif BUILDING_LIBGUILE && (defined _WIN32 || defined __CYGWIN__)
+#elif defined BUILDING_LIBGUILE && (defined _WIN32 || defined __CYGWIN__)
 # define SCM_API __declspec(dllexport) extern
 #elif defined _WIN32 || defined __CYGWIN__
 # define SCM_API __declspec(dllimport) extern
-- 
2.45.2

